### PR TITLE
Support multiple embeds

### LINKED ISSUE #132: Support multiple embeds in Message#edit
Title

### PR DESCRIPTION
**Pull request by Dakurei**
# Summary

Fixes the problem preventing the creation and editing of messages with multiple embeds

(Based on PR #193)

---

## Fixed
Fix #132

### CODE REVIEW

File: lib/discordrb/data/webhook.rb [hunk 1]
```diff
@@ -84,9 +84,9 @@ def name=(name)
     def update(data)
       # Only pass a value for avatar if the key is defined as sending nil will delete the
       data[:avatar] = avatarise(data[:avatar]) if data.key?(:avatar)
-      data[:channel_id] = data[:channel].resolve_id
+      data[:channel_id] = data[:channel]&.resolve_id
       data.delete(:channel)
-      update_webhook(data)
+      update_webhook(**data)
```
Comment (swarley) [inline lib/discordrb/data/webhook.rb:89]: why is data being spread here?

File: lib/discordrb/data/component.rb [hunk 2]
```diff
@@ -14,7 +14,7 @@ def self.from_data(data, bot)
         ActionRow.new(data, bot)
       when Webhooks::View::COMPONENT_TYPES[:button]
         Button.new(data, bot)
-      when Webhooks::View::COMPONENT_TYPES[:select_menu]
+      when Webhooks::View::COMPONENT_TYPES[:string_select]
```
Comment (swarley) [inline lib/discordrb/data/component.rb:17]: Are we not supporting the other select types?

File: lib/discordrb/events/interactions.rb [hunk 3]
```diff
@@ -381,28 +402,77 @@ def value(custom_id)
...
   end
 
-  # Event handler for a Button interaction event.
-  class ButtonEventHandler < ComponentEventHandler
+  # Event handler for a modal submission.
+  class ModalSubmitEventHandler < ComponentEventHandler
   end
 
-  # Event for when a user interacts with a select menu component.
-  class SelectMenuEvent < ComponentEvent
```
Comment (swarley) [inline lib/discordrb/events/interactions.rb:389]: This should not be removed and should be the base for all other select events

File: lib/discordrb/events/interactions.rb [hunk 4]
```diff
@@ -381,28 +402,77 @@ def value(custom_id)
...
     def initialize(data, bot)
       super
 
-      @values = data['data']['values']
+      @values = data['data']['values'].map { |e| bot.user(e) }
     end
   end
 
-  # Event handler for a select menu component.
-  class SelectMenuEventHandler < ComponentEventHandler
```
Comment (swarley) [inline lib/discordrb/events/interactions.rb:402]: Likewise this should be the base for other select menu event handlers

File: spec/api/channel_spec.rb [hunk 5]
```diff
@@ -55,7 +55,7 @@
 
   describe '.delete_all_emoji_reactions' do
     let(:message_id) { double('message_id', to_s: 'message_id') }
-    let(:emoji) { "\u{1F525}" }
+    let(:emoji) { '\u{1F525}' }
```
Comment (swarley) [inline spec/api/channel_spec.rb:58]: Needs to be double quotes

File: lib/discordrb/data/webhook.rb (hunk 1 above)
Comment (Dakurei) [reply lib/discordrb/data/webhook.rb:89]: I had noticed that this is a problem that affects Ruby 3+.
Without this, the test suite concerning webhooks does not pass, at least without modifying the test directly

**Comment by swarley** [comment]
Closed in favor of #209

**Pull request closed without merging**